envoy/formatter/http_formatter_context.h

```diff
@@ -22,12 +24,14 @@ class HttpFormatterContext {
    * @param request_headers supplies the request headers.
    * @param response_headers supplies the response headers.
    * @param response_trailers supplies the response trailers.
+   * @param active_span supplies the active span.
    * @param local_reply_body supplies the local reply body.
    * @param log_type supplies the access log type.
    */
   HttpFormatterContext(const Http::RequestHeaderMap* request_headers = nullptr,
                        const Http::ResponseHeaderMap* response_headers = nullptr,
                        const Http::ResponseTrailerMap* response_trailers = nullptr,
+                       const Tracing::Span& active_span = Tracing::NullSpan::instance(),
                        absl::string_view local_reply_body = {},
                        AccessLogType log_type = AccessLogType::NotSet);
```

Comment:
I think it would be better to take a pointer as parameter with default value nullptr. And add the new parameter at the end of parameter list.
Then, most of changes in this PR would be reverted.

Reply:
sg